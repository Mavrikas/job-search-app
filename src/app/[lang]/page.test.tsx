import React from 'react';
import { render } from '@testing-library/react';
import Home from './page';

// Mock child components
jest.mock('../components/JobsList/JobsList', () => ({
  JobsList: (props: any) => <div data-testid="jobs-list">{JSON.stringify(props)}</div>,
}));
jest.mock('../components/Search/Search', () => ({
  Search: () => <div data-testid="search" />,
}));
jest.mock('../components/ActionBar/ActionBar', () => ({
  ActionBar: (props: any) => <div data-testid="action-bar">{JSON.stringify(props)}</div>,
}));

// Mock constants and types
jest.mock('@/store/constants', () => ({
  URL: 'http://mocked-url/',
}));
type JobFilters = { categories: string[]; locations: string[] };
jest.mock('@/store/types', () => ({}));

// Mock fetch
global.fetch = jest.fn();

const mockFilters: JobFilters = {
  categories: ['Engineering', 'Design'],
  locations: ['Remote', 'Onsite'],
};

describe('Home', () => {
  beforeEach(() => {
    (fetch as jest.Mock).mockResolvedValue({
      json: jest.fn().mockResolvedValue(mockFilters),
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders with default parameters', async () => {
    const params = Promise.resolve({ lang: 'en', page: 1 });
    const searchParams = Promise.resolve({});
    const { asFragment } = render(await Home({ params, searchParams }));
    expect(asFragment()).toMatchSnapshot();
  });

  it('passes search parameters to JobsList', async () => {
    const params = Promise.resolve({ lang: 'el', page: 2 });
    const searchParams = Promise.resolve({
      query: 'react',
      location: 'Remote',
      category: 'Engineering',
      page: '3',
      pageSize: '20',
    });
    const { asFragment } = render(await Home({ params, searchParams }));
    expect(asFragment()).toMatchSnapshot();
  });

  it('fetches filters from the correct URL', async () => {
    const params = Promise.resolve({ lang: 'en', page: 1 });
    const searchParams = Promise.resolve({});
    await Home({ params, searchParams });
    expect(fetch).toHaveBeenCalledWith('http://mocked-url/jobs/filters');
  });
});
