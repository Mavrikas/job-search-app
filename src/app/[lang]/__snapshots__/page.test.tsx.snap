// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home passes search parameters to JobsList 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center justify-center min-h-screen p-8 pb-20 gap-8 sm:p-20 font-[family-name:var(--font-geist-sans)]"
  >
    <div
      data-testid="action-bar"
    >
      {"filters":{"categories":["Engineering","Design"],"locations":["Remote","Onsite"]},"lang":"el"}
    </div>
    <main
      class="min-h-screen flex flex-col items-center justify-start w-full max-w-[800px] gap-1"
    >
      <div
        data-testid="search"
      />
      <div
        data-testid="jobs-list"
      >
        {"lang":"el","page":3,"query":"react","location":"Remote","category":"Engineering","pageSize":20}
      </div>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Home renders with default parameters 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center justify-center min-h-screen p-8 pb-20 gap-8 sm:p-20 font-[family-name:var(--font-geist-sans)]"
  >
    <div
      data-testid="action-bar"
    >
      {"filters":{"categories":["Engineering","Design"],"locations":["Remote","Onsite"]},"lang":"en"}
    </div>
    <main
      class="min-h-screen flex flex-col items-center justify-start w-full max-w-[800px] gap-1"
    >
      <div
        data-testid="search"
      />
      <div
        data-testid="jobs-list"
      >
        {"lang":"en","page":1,"query":"","location":"","category":"","pageSize":10}
      </div>
    </main>
  </div>
</DocumentFragment>
`;
